src/Sylius/Component/Variation/spec/Generator/VariantGeneratorSpec.php

```diff
@@ -84,14 +93,62 @@ function it_generates_variants_for_every_value_of_an_objects_single_option(
         $this->generate($productVariable);
     }
 
+    function it_does_not_generate_variant_if_given_variant_exist(
+        VariableInterface $productVariable,
+        FactoryInterface $variantFactory,
+        SetBuilderInterface $setBuilder,
+        OptionInterface $colorOption,
+        OptionValue $blackColor,
+        OptionValue $whiteColor,
+        OptionValue $redColor,
+        VariantInterface $masterVariant,
+        VariantInterface $permutationVariant,
+        VariantsParityCheckerInterface $variantsParityChecker
+    ) {
+        $productVariable->hasOptions()->willReturn(true);
+
+        $productVariable->getOptions()->willReturn([$colorOption]);
+
+        $colorOption->getValues()->willReturn([$blackColor, $whiteColor, $redColor]);
+
+        // Stubbing `OptionValue` instead of `OptionValueInterface` in order to stub `getId` method.
```

Comment:
`OptionValueInterface` extends `ResourceInterface` which has `getId` method defined. You should use interface.